types/pull-stream/pull-stream-tests.ts

```diff
@@ -4,99 +4,133 @@ import pull = require('pull-stream');
  * pull *
  ********/
 
-let source: pull.Source<string> = (end, cb) => undefined;
-let through: pull.Through<string, string> = (source) => source;
-let sink: pull.Sink<string> = (source) => undefined;
+let source: pull.Source<string> = (end, cb) => {};
+let through: pull.Through<string, string> = source => source;
+let sink: pull.Sink<string> = source => {};
+
+function voidFunc(): void {}
+
+type Void = ReturnType<typeof voidFunc>;
 
 // Start with source and end with sink
-let nothing: undefined;
+let nothing: Void;
 nothing = pull();
 nothing = pull(source, sink);
 nothing = pull(source, through, sink);
 nothing = pull(source, through, through, sink);
 
-// End with sink
-sink = pull(sink);
-sink = pull(through, sink);
-sink = pull(through, through, sink);
-
 // Start with source
 source = pull(source);
 source = pull(source, through);
 source = pull(source, through, through);
 
+// End with sink
+sink = pull(sink);
+sink = pull(through, sink);
+sink = pull(through, through, sink);
+
 // Through only
 through = pull(through);
 through = pull(through, through);
 through = pull(through, through, through);
 
-const numberSource: pull.Source<number> = (end, cb) => undefined;
-const parseNumber: pull.Through<string, number> = (source) => numberSource;
-const numberSink: pull.Sink<number> = (source) => undefined;
+const numberSource: pull.Source<number> = (end, cb) => {};
+const parseNumber: pull.Through<string, number> = source => numberSource;
+const numberSink: pull.Sink<number> = source => {};
 
 // Strictly typed pipe
 nothing = pull(source, through, parseNumber, numberSink);
 
 // Long pipe
-let result: pull.Source<any> | pull.Sink<any> | pull.Through<any, any> | undefined;
+let result: pull.Source<any> | pull.Sink<any> | pull.Through<any, any> | Void;
 result = pull(source, through, through, through, through, through, through, through, through, sink);
 
 /***********
  * sources *
  ***********/
 
 // count
-nothing = pull(pull.count(), numberSink);
-nothing = pull(pull.count(5), numberSink);
-nothing = pull(pull.count(5, (err) => { if (err) err.stack; }), numberSink);
+pull(pull.count(), numberSink);
+pull(pull.count(5), numberSink);
 
 // empty
 nothing = pull(pull.empty(), sink);
 
 // error
-nothing = pull(pull.error(), sink);
+nothing = pull(pull.error(new Error()), sink);
 
 // infinite
 nothing = pull(pull.infinite(), numberSink);
-nothing = pull(pull.infinite(() => 'value'), sink);
-nothing = pull(pull.infinite(() => 'value', (err) => { if (err) err.stack; }), sink);
+nothing = pull(
+    pull.infinite(() => 'value'),
+    sink,
+);
+nothing = pull(
+    pull.infinite(() => 'value'),
+    sink,
+);
```

Comment:
Isn't this the exact same statement as the one above? You can probably remove it.

Reply:
<img width="900" alt="image" src="https://user-images.githubusercontent.com/1164378/90304558-bdba3780-deeb-11ea-80b6-0671159c2572.png">
Because `void` can only be used for return types.

Reply:
yes, I'll remove it.